cartoframes/data/dataset.py

```diff
@@ -504,34 +506,44 @@ def delete(self):
     def exists(self):
         """Checks to see if table exists"""
         try:
-            self._con.sql_client.send(
+            self._client.execute_query(
                 'EXPLAIN SELECT * FROM "{table_name}"'.format(table_name=self._table_name),
                 do_post=False)
             return True
+        except CartoRateLimitException as err:
+            raise err
         except CartoException as err:
             # If table doesn't exist, we get an error from the SQL API
             self._con._debug_print(err=err)
             return False
 
     def is_public(self):
         """Checks to see if table or table used by query has public privacy"""
-        public_context = get_public_context(self.context)
+        public_client = get_client_with_public_creds(self.context)
         try:
-            public_context.sql_client.send('EXPLAIN {}'.format(self.get_query()), do_post=False)
+            public_client.execute_query('EXPLAIN {}'.format(self.get_query()), do_post=False)
             return True
+        except CartoRateLimitException as err:
+            raise err
         except CartoException:
             return False
 
+    def _get_client(self):
+        if self._con:
+            return get_client(self._con.creds, self._con.session)
```

Comment:
I would raise an error here if the client cannot be initialized

Reply:
When we are working with local objects (DataFrames or GeoDataFrames) we don't have context and it is right. So, we don't have to return an error here.

When the context is added, we are calling the get_client function again